snapcraft/internal/errors.py

```diff
@@ -202,14 +202,36 @@ def __init__(self, remote):
         super().__init__(remote=remote)
 
 
-class InvalidDesktopFileError(SnapcraftError):
+class MissingDesktopFileError(SnapcraftError):
 
     fmt = (
         'Failed to generate desktop file: '
-        'Invalid desktop file {filename!r}: {message}.'
-        # FIXME include how to fix each of the possible desktop file errors.
-        # https://bugs.launchpad.net/snapcraft/+bug/1727435
-        # --elopio - 2017-10-25
+        'Invalid desktop file {filename!r}: Check if desktop file matches app name or Create a new path.'
+        
+    )
+
+    def __init__(self, filename, message):
+        super().__init__(filename=filename, message=message)
+
+
+class MissingDesktopSectionError(SnapcraftError):
+
+    fmt = (
+        'Failed to find any Desktop Entry: '
+        'No Desktop Entry for {filename!r}: Check in home directory that .desktop file exists.'
+        
+    )
+
+    def __init__(self, filename, message):
+        super().__init__(filename=filename, message=message)
+
+
+class MissingDesktopExecError(SnapcraftError):
+
+    fmt = (
+        'Failed to find Exec line: '
+        'No Exec line for {filename!r}: Check if file exists or not.'
```

Comment:
Again, the file existence is not at question in this error. Make this `Missing 'Exec' key in desktop file {filename!r}`.